### PR TITLE
BACKLOG-13003 Collapse primary nav when an item is clicked

### PR DESCRIPTION
**Pull request by AKarmanov**
## JIRA

https://jira.jahia.org/browse/BACKLOG-13003

## Description

Collapse primary nav when an item is clicked

### CODE REVIEW

File: src/components/PrimaryNav/PrimaryNav.jsx [hunk 1]
```diff
@@ -63,15 +63,17 @@ NavHeader.propTypes = {
     headerLogo: PropTypes.node
 };
 
-export const PrimaryNav = ({headerLogo, top, bottom, headerCaption, modeIcon, ...props}) => {
+export const PrimaryNav = ({headerLogo, top, bottom, headerCaption, modeIcon, isCollapsibleOnCLick, ...props}) => {
```
Comment (ffffffelix) [inline src/components/PrimaryNav/PrimaryNav.jsx:66]: Do we want to keep the choice collapsible on click or not?
I think we can force the collapse in all cases.

If we keep it, please fix the typo `isCollapsibleOnClick`

File: src/components/PrimaryNav/PrimaryNavItem/PrimaryNavItem.jsx [hunk 2]
```diff
@@ -62,23 +73,35 @@ ItemTypeResolver.propTypes = {
...
+        <li
+            className={classnames(
+                styles.primaryNavItem,
+                {[styles.selected]: isSelected},
+                className
+            )}
+            title={label}
+            onClick={e => {
+                primaryNavContext.collapse();
+                if (onClick) {
```
Comment (ffffffelix) [inline src/components/PrimaryNav/PrimaryNavItem/PrimaryNavItem.jsx:89]: You can add this, to manage empty onClick props
```
PrimaryNavItem.defaultProps = {
    onClick: () => {}
}
```

File: src/components/PrimaryNav/PrimaryNav.jsx (hunk 1 above)
Comment (AKarmanov) [reply src/components/PrimaryNav/PrimaryNav.jsx]: My original idea was to make this behaviour optional but seeing how contested it is and that there is no clear direction for what  should be done and given that not having it means that I won't have to modify any other module I decided to remove it :)

File: src/components/PrimaryNav/PrimaryNavItem/PrimaryNavItem.jsx (hunk 2 above)
Comment (AKarmanov) [reply src/components/PrimaryNav/PrimaryNavItem/PrimaryNavItem.jsx]: good point

**Pull request merged**